File: fixtures/scale/dart/large/lib/util/block_codec.dart
import 'dart:typed_data';

/// Big-endian 64-bit block packing shared by the block ciphers.
int readBlock(Uint8List data, int at) {
  var value = 0;
  for (var i = 0; i < 8; i++) {
    value = (value << 8) | data[at + i];
  }
  return value;
}

void writeBlock(Uint8List data, int at, int value) {
  var v = value;
  for (var i = 7; i >= 0; i--) {
    data[at + i] = v & 0xFF;
    v >>= 8;
  }
}

File: fixtures/scale/scala/large/src/cryptobench/util/BlockCodec.scala
package cryptobench.util

/** Big-endian 64-bit block packing shared by the block ciphers. */
object BlockCodec {
  def read(data: Array[Byte], at: Int): Long = {
    var value = 0L
    for (i <- 0 until 8) {
      value = (value << 8) | (data(at + i).toLong & 0xFF)
    }
    value
  }

  def write(data: Array[Byte], at: Int, value: Long): Unit = {
    var v = value
    for (i <- 7 to 0 by -1) {
      data(at + i) = v.toByte
      v = v >>> 8
    }
  }
}
